### DISCUSSION TITLE
Function calling with Mixtral 8x22B

### DISCUSSION THREAD

**Discussion by deepinderdeol** [microsoft/semantic-kernel#5970] [Q&A] [answered]
Mixtral 8x22B has function calling capability. Is it possible to use this model with Semantic-Kernel? 
Ollama also has this model but will auto-invoke etc. work the same as with GPT4?

If anyone has tried this out then can you please provide some guidance?

Thanks.

**Comment by markwallace-microsoft** [comment, marked as answer]
Hi @deepinderdeol 
we have a sample which shows how to use our OpenAI connector with other models, see: https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/KernelSyntaxExamples/Example88_CustomMessageAPIEndpoint.cs

Please try this out and let us know how things go.

**Reply by Natfan** [reply]
No, you don't. That file has been deleted.

**Reply by dmytrostruk** [reply]
Hi @Natfan , thanks for pointing that out, we restructured our samples recently, so this file was moved to another place.
Here is an example of using function calling for Mistral AI models:
https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/Concepts/ChatCompletion/MistralAI_FunctionCalling.cs

This example shows a way of using OpenAI connector with other APIs that supports the same ChatCompletion Message API standard from OpenAI:
https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/Concepts/LocalModels/MultipleProviders_ChatCompletion.cs
